cli/command/container/cp.go

```diff
@@ -34,11 +39,91 @@ const (
 type cpConfig struct {
 	followLink bool
 	copyUIDGID bool
+	quiet      bool
 	sourcePath string
 	destPath   string
 	container  string
 }
 
+//CopyReader is a Wrapper to content when it is a Reader
+type CopyReader struct {
```

Comment:
Do these reader types need to be exported?

Reply:
Also can this just be one type?
For the case where we don't have a closer we can wrap it in `ioutil.NopCloser`

Reply:
The types don't need to be exported. We can made then private.
I have created two types because in the `copyFromContainer` method the parameter that will be copied (`content`) is a `ReadCloser` type (returned by client.CopyFromContainer), and in the method `copyToContainer` the `content` parameter is a `Reader`.
Do you have any suggestions to change this?

Reply:
I would always have it take a ReadCloser and wrap the reader in a no-op closer.

Reply:
Ok, I'll check this.